wcfsetup/install/files/lib/system/cache/tolerant/TagCloudCache.class.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace wcf\system\cache\tolerant;
+
+use wcf\data\tag\Tag;
+use wcf\data\tag\TagCloudTag;
+use wcf\system\database\util\PreparedStatementConditionBuilder;
+use wcf\system\WCF;
+
+/**
+ * Caches the tag cloud.
+ *
+ * @author Olaf Braun
+ * @copyright 2001-2025 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.2
+ *
+ * @extends AbstractTolerantCache<array<string, TagCloudTag>>
+ */
+class TagCloudCache extends AbstractTolerantCache
+{
+    public function __construct(
+        /** @var int[] */
+        public readonly array $objectTypeIDs,
+        /** @var int[] */
+        public readonly array $languageIDs = [],
+    ) {
+    }
+
+    private static function compareTags(TagCloudTag $tagA, TagCloudTag $tagB): int
+    {
+        if ($tagA->counter > $tagB->counter) {
+            return -1;
+        }
+        if ($tagA->counter < $tagB->counter) {
+            return 1;
+        }
+
+        return 0;
+    }
+
+    #[\Override]
+    public function getLifetime(): int
+    {
+        return 3_600;
+    }
+
+    #[\Override]
+    protected function rebuildCacheData(): array
+    {
+        if ($this->objectTypeIDs === []) {
+            return [];
+        }
+
+        return $this->fetchTags($this->fetchTagIDs());
+    }
+
+    /**
+     * @param list<int> $tagIDs
+     *
```

Comment:
```suggestion
     * @param array<int, int> $tagIDs
```

A `list<…>` is a zero-based numeric array without gaps. The return value of `fetchTagIDs()` is a map, not a list anyway. Despite that, `\array_keys()` in line 70 would have be an odd choice for a zero-based array.